From 108f8191106401f1b325f8c0c1ae5e0efe27dcb4 Mon Sep 17 00:00:00 2001
From: Daniel Vanzo <daniel.vanzo@external.total.com>
Date: Sun, 2 Jul 2023 16:39:10 +0200
Subject: [PATCH] Remove use of depreciated numpy API for Compatibility with
 new numpy version >= 1.24

---
 f90wrap/arraydatamodule.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/f90wrap/arraydatamodule.c b/f90wrap/arraydatamodule.c
index 73e56e7..5dba296 100644
--- a/f90wrap/arraydatamodule.c
+++ b/f90wrap/arraydatamodule.c
@@ -59,13 +59,13 @@ get_array(PyObject *self, PyObject *args)
   /* Processing variable this */
   this_Dims[0]=sizeof_fortran_t;
   capi_this_intent |= F2PY_INTENT_IN;
-  capi_this_tmp = array_from_pyobj(PyArray_INT,this_Dims,this_Rank,capi_this_intent,this_capi);
+  capi_this_tmp = array_from_pyobj(NPY_INT,this_Dims,this_Rank,capi_this_intent,this_capi);
   if (capi_this_tmp == NULL) {
     if (!PyErr_Occurred())
       PyErr_SetString(PyExc_TypeError,"failed in converting 1st argument `this' of get_array to C/Fortran array" );
     goto fail;
   } else {
-    this = (int *)(capi_this_tmp->data);
+    this = (int *)(PyArray_DATA(capi_this_tmp));
   }
 
   /* Processing variable arrayfunc */
@@ -107,7 +107,7 @@ get_array(PyObject *self, PyObject *args)
   /* Construct array */
   descr = PyArray_DescrNewFromType(typenum);
   array = (PyArrayObject*) PyArray_NewFromDescr(&PyArray_Type, descr, nd, dimensions, NULL, 
-                                                data, NPY_FORTRAN | NPY_WRITEABLE | NPY_ALIGNED, NULL);
+                                                data, NPY_ARRAY_F_CONTIGUOUS | NPY_ARRAY_WRITEABLE | NPY_ARRAY_ALIGNED, NULL);
   free(dimensions);
   if((PyObject *)capi_this_tmp!=this_capi) {
     Py_XDECREF(capi_this_tmp);
